modules/renter/upload_test.go

```diff
@@ -0,0 +1,33 @@
+package renter
+
+import (
+	"testing"
+)
+
+// TestRenterSiapathValidate verifies that the validateSiapath function correctly validates SiaPaths.
+func TestRenterSiapathValidate(t *testing.T) {
+	var pathtests = []struct {
+		in    string
+		valid bool
+	}{
+		{"valid/siapath", true},
+		{"../../../directory/traversal", false},
+		{"testpath", true},
+		{"valid/siapath/../with/directory/traversal", false},
```

Comment:
If we allow some directory traversal, it opens up the possibility of users inadvertantly overwriting their siafiles, take for example:

`folder/siafile.png`
`folder/folder2/../siafile.png`

these two siapaths resolve on the filesystem to the same file and would cause an overwrite condition

I think it's safer just to disallow directory traversal entirely, I don't see a good reason to have it.